src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.cs

```diff
@@ -10,28 +10,40 @@ namespace System.Text.Json
 {
     internal static partial class JsonWriterHelper
     {
-        public static void WriteIndentation(Span<byte> buffer, int indent)
+        public static void WriteIndentation(Span<byte> buffer, int indent, byte indentByte)
         {
-            Debug.Assert(indent % JsonConstants.SpacesPerIndent == 0);
             Debug.Assert(buffer.Length >= indent);
 
             // Based on perf tests, the break-even point where vectorized Fill is faster
             // than explicitly writing the space in a loop is 8.
-            if (indent < 8)
+            if (indent is < 8)
             {
                 int i = 0;
                 while (i < indent)
                 {
-                    buffer[i++] = JsonConstants.Space;
-                    buffer[i++] = JsonConstants.Space;
+                    buffer[i++] = indentByte;
```

Comment:
The previous logic was doing loop unrolling in chunks of two. You could keep that logic by doing something like
```C#
while (i + 1 < indent)
{
    buffer[i++] = indentByte;
    buffer[i++] = indentByte;
}

if (i < indent)
{
    buffer[i] = indentByte;
}
```

This should also cover the case where `indent == 0`

Reply:
I like this proposal.